contracts/modules/cover/CoverNFTDescriptor.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-v4/utils/Strings.sol";
+import "@openzeppelin/contracts-v4/utils/Base64.sol";
+import "../../interfaces/ICover.sol";
+import "../../interfaces/ICoverNFTDescriptor.sol";
+import "../../libraries/DateTime.sol";
+import "./CoverNFT.sol";
+
+contract CoverNFTDescriptor {
+  using Strings for uint256;
+  using DateTime for uint256;
+
+  string[] public coverAssets;
+
+  constructor() {
+    // TODO: how to handle asset symbols?
+    coverAssets = ["WEI", "DAI"];
+  }
+
+  function tokenURI(CoverTokenURIParams calldata params) public view returns (string memory) {
+
+    string memory image = Base64.encode(bytes(generateSVGImage(params)));
+
+    return string(
+      abi.encodePacked(
+        "data:application/json;base64,",
+        Base64.encode(
+          bytes(
+            abi.encodePacked(
+              '{"name":"', params.name, '",',
+              '"description":"', generateDescription(params), '",',
+              '"image": "',
+              'data:image/svg+xml;base64,',
+              image,
+              '"}'
+            )
+          )
+        )
+      )
+    );
+  }
+
+  function generateDescription(CoverTokenURIParams calldata params) public view returns (string memory) {
+
+    CoverNFT coverNFT = CoverNFT(msg.sender);
+    ICover cover = ICover(coverNFT.operator());
```

Comment:
Fetch the cover address from master instead, as the operator is not guaranteed to be the Cover contract in the future.